### ISSUE TITLE
Installing 2.2.1 requires ruby 2.2.2

### ISSUE THREAD

**Issue by njam** [rodjek/librarian-puppet#330]
Due to:
> activesupport requires Ruby version >= 2.2.2

Turns out:
- A: `librarian-puppet 2.2.1` requires `puppet_forge ~> 1.0`
- B: `puppet_forge 1.0.5` requires `her ~> 0.6.8` ([reference](https://github.com/puppetlabs/forge-ruby/blob/v1.0.5/puppet_forge.gemspec))
- C: `her 0.6.8` requires `activesupport  >= 3.0.0` ([reference](https://github.com/remiprev/her/blob/v0.6.8/her.gemspec))
- D: `activesupport 5.0.0.beta1` requires `ruby >= 2.2.2` ([reference](https://github.com/rails/rails/blob/v5.0.0.beta1/activesupport/activesupport.gemspec))

C was fixed in https://github.com/remiprev/her/issues/227, but not for version 0.6 which is very old.

Best would probably be to change A, and upgrade to puppet_forge 2?
Wdyt?

**Comment by njam** [comment]
As a workaround you can install `activesupport` version `4.2.5`.
cc @richdel

**Comment by breser** [comment]
activesupport 5.0.0 is now released.  So now I'm seeing this issue with installing 2.2.1, didn't see it a few weeks ago despite the age of this ticket, but I presume that is because when this ticket was opened it was just a beta.

As a pointer to others who might be running into this issue.  Just installing activesupport 4.2.5 (as mentioned in a previous comment) didn't help me.  You need to install all the dependencies for librarian-puppet individual and use `--ignore-dependencies` on any of the ones that are in the chain that requires activesupport.  You might be able to get around that with `--minimal-deps` but the version of gem I had didn't have that option.

But replacing: `gem install librarian-puppet -v 2.2.1` with the following set of commands resolved the problem for me:

```
gem install activesupport -v 4.2.6
gem install activemodel -v 4.2.6
gem install faraday -v 0.9.2
gem install multi_json -v 1.12.1
gem install her -v 0.6.8 --ignore-dependencies
gem install puppet_forge -v 1.0.5 --ignore-dependencies
gem install librarianp -v 0.6.3
gem install rsync -v 1.0.9
gem install librarian-puppet -v 2.2.1 --ignore-dependencies
```

This is of course really ugly and prone to breakage as other dependency versions upgrade.  The specific versions I've chosen here match what I was getting before activesupport released 5.0.0.  I'm also not installing any development dependencies.  So if you want those use `gem dependency` to figure them out and add them to the list.

**Comment by njam** [comment]
I _think_ this should be solved by https://github.com/voxpupuli/librarian-puppet/pull/1

Note this project moved to a different repo, and the lastest version is 2.2.3:
https://github.com/voxpupuli/librarian-puppet/releases
